src/filesystem/drive/directory_handle.rs

```diff
@@ -519,61 +521,78 @@ impl DirectoryHandle {
 
         drop(inner_read);
 
-        let (parent_path, name) = path.split_at(path.len() - 1);
-        let file_name = NodeName::try_from(name[0]).map_err(OperationError::InvalidName)?;
-
-        tracing::info!(?path, ?file_name, "drive::write");
-
-        let parent_id = match walk_path(&self.inner, self.cwd_id, parent_path, 0).await? {
-            WalkState::FoundNode { node_id } => node_id,
-            WalkState::MissingComponent { .. } => return Err(OperationError::PathNotFound),
+        let existing_file = match walk_path(&self.inner, self.cwd_id, path, 0).await {
+            Ok(WalkState::FoundNode { node_id }) => Some(node_id),
+            _ => None,
         };
 
-        tracing::info!(?parent_id, "drive::write::parent_id");
-
-        let inner_read = self.inner.read().await;
-        let parent_node = inner_read.by_id(parent_id)?;
-        let parent_perm_id = parent_node.permanent_id();
-        drop(inner_read);
-
         let data_size = data.len() as u64;
-        let node_name = file_name.clone();
-
-        // todo(sstelfox): handle the special case of an empty file, it shouldn't be a stub and
-        // doesn't need to go through the encoding process.
-
-        let new_permanent_id = self
-            .insert_node(
-                rng,
-                parent_perm_id,
-                |rng, new_node_id, parent_id, actor_id| async move {
-                    NodeBuilder::file(node_name)
-                        .with_parent(parent_id)
-                        .with_id(new_node_id)
-                        .with_owner(actor_id)
-                        .with_size_hint(data_size)
-                        .build(rng)
-                        .map_err(OperationError::CreationFailed)
-                },
-            )
-            .await?;
+        let new_permanent_id = match existing_file {
+            Some(existing_file) => {
+                let inner_read = self.inner.read().await;
+                let node = inner_read.by_id(existing_file)?;
+                let permanent_id = node.permanent_id();
+                drop(inner_read);
```

Comment:
Nit: I suspect this explicit drop isn't necessary since it immediately returns and the var is defined in this scope, but sometime autodrop is weird

Reply:
Ah good catch